blaze-client/src/main/scala/org/http4s/client/blaze/PooledHttp1Client.scala

```diff
@@ -8,18 +8,21 @@ import cats.effect._
 object PooledHttp1Client {
   private val DefaultMaxTotalConnections = 10
   private val DefaultMaxWaitQueueLimit = 256
+  private val DefaultWaitExpiryTime = -1
 
   /** Construct a new PooledHttp1Client
     *
     * @param maxTotalConnections maximum connections the client will have at any specific time
     * @param maxWaitQueueLimit maximum number requests waiting for a connection at any specific time
     * @param maxConnectionsPerRequestKey Map of RequestKey to number of max connections
+    * @param waitExpiryTime timeout duration for request waiting in queue, default value -1, never expire
     * @param config blaze client configuration options
     */
   def apply[F[_]: Effect](
       maxTotalConnections: Int = DefaultMaxTotalConnections,
       maxWaitQueueLimit: Int = DefaultMaxWaitQueueLimit,
       maxConnectionsPerRequestKey: RequestKey => Int = _ => DefaultMaxTotalConnections,
+      waitExpiryTime: RequestKey => Int = _ => DefaultWaitExpiryTime,
```

Comment:
How about a `Duration` for the type here instead of `Int`?

Reply:
Ok.